src/MonitoringFunctions.Common/DataService/IDataService.cs

```diff
@@ -17,6 +18,14 @@ internal interface IDataService : IDisposable
         /// <returns>A task, tracking the initiated async operation. Errors should be reported through exceptions.</returns>
         Task ReportUrlAccessAsync(string monitorName, HttpResponseMessage httpResponse, CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Stores the details of the <see cref="HttpRequestLogEntry"/> in the underlying data store.
+        /// </summary>
+        /// <param name="httpRequestLogEntry"><see cref="HttpRequestLogEntry"/> to be stored.</param>
+        /// <param name="cancellationToken">Optional <see cref="CancellationToken"/> instance.</param>
+        /// <returns>A task, tracking the initiated async operation. Errors should be reported through exceptions.</returns>
+        Task ReportUrlAccessAsync(HttpRequestLogEntry httpRequestLogEntry, CancellationToken cancellationToken = default);
```

Comment:
Should we deprecate/remove the overload that accepts an `HttpResponseMessage` soon while we have the chance?